src/xrpld/overlay/detail/PeerImp.cpp

```diff
@@ -241,9 +240,11 @@ PeerImp::send(std::shared_ptr<Message> const& m)
 {
     if (!strand_.running_in_this_thread())
         return post(strand_, std::bind(&PeerImp::send, shared_from_this(), m));
+
     if (gracefulClose_)
         return;
-    if (detaching_)
```

Comment:
Detaching  was there to indicate that the connection is already in the process of being terminated. 

To prevent writing to an already closed socket, I added an `is_open` check.

However, even if we fail, and end up writing a message to socket that is in the process of being closed, it will caught in the `onWriteMessage` method, by either another `is_open` check, or an error `boost::asio::error::operation_aborted`.

When `shutdown` or `close` is called, any I/O operations are aborted with the `boost::asio::error::operation_aborted` code.